### PR TITLE
Feature : Question (entity, IRepository, Repository, Exception.Question, QuestionRepositoryTest, documentation)

### PR DESCRIPTION
**Pull request by 3l5a**
Lel

### CODE REVIEW

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    /// <summary>
+    /// Delete a question and its fields.
+    /// </summary>
+    /// <param name="id"></param>
+    Task DeleteAsync(int id);
+
+    /// <summary>
+    /// Get all questions of a form with their fields.
+    /// </summary>
+    /// <returns>List of questions</returns>
```
Comment (htkisui) [inline QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: "." MISSING!!!!!

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    Task<List<Question>> GetAllAsync();
+
+    /// <summary>
+    /// Get a question by Id with its fields.
+    /// </summary>
+    /// <param name="id"></param>
+    /// <returns>Question</returns>
```
Comment (htkisui) [inline QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: "." !!!!!!!!!

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    Task<Question> GetByIdAsync(int id);
+
+    /// <summary>
+    /// Update a question
```
Comment (htkisui) [inline QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: ...........................

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    /// </summary>
+    /// <param name="question"></param>
+    /// <returns>Question updated</returns>
```
Comment (htkisui) [inline QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: .

File: QuikForm.Repository.Contracts/Exceptions/Questions/QuestionNotFoundException.cs [hunk 5]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace QuikForm.Repository.Contracts.Exceptions.Questions;
+public class QuestionNotFoundException : Exception
+{
+    public override string Message => "Question not found";
```
Comment (htkisui) [inline QuikForm.Repository.Contracts/Exceptions/Questions/QuestionNotFoundException.cs]: Question is not found.

File: QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,175 @@
...
+using System.Threading.Tasks;
+
+namespace QuikForm.Repositories.Repositories.Tests;
+
+[TestClass()]
+public class QuestionRepositoryTests
+{
+    #region CreateAsync
+    [TestMethod()]
+    public async Task CreateAsync_QuestionToAdd_QuestionWithId()
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs]: CreateAsync_QuestionToAdd_QuestionAdded

File: QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    #region DeleteAsync
+    [TestMethod()]
+    public async Task DeleteAsync_QuestionToDelete_EmptyList()
+    {
+        // Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        QuestionRepository questionRepository = new QuestionRepository(context);
+        Question qToDelete = new Question() { Id = 1, Label = "Label" };
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs]: questionToDelete or move in AddAsync inside

File: QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        await context.Questions.AddAsync(new Question { Id = 1, Label = "Label", IsMandatory = false });
+        await context.SaveChangesAsync();
+
+        Question questionExpected = new Question() { Id = 1, Label = "LabelUpdated", IsMandatory = true };
+
+        //Act
+        Question questionUpdated = await questionRepository.UpdateAsync(questionExpected);
+
+        //Assert
+        ApplicationDbContext otherContext = new ApplicationDbContext(builder.Options);
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs:156]: move otherContext to Act section

File: QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs [hunk 9]
```diff
@@ -0,0 +1,175 @@
...
+    public async Task UpdateAsync_InvalidQuestionToUpdate_ThrowQuestionNotFoundException()
+    {
+        //Arrange
+        DbContextOptionsBuilder<ApplicationDbContext> builder = new DbContextOptionsBuilder<ApplicationDbContext>().UseInMemoryDatabase("QuikFormTest");
+        ApplicationDbContext context = new ApplicationDbContext(builder.Options);
+        context.Database.EnsureDeleted();
+        QuestionRepository questionRepository = new QuestionRepository(context);
+
+        //Act
+        Question question = new Question();
```
Comment (htkisui) [inline QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs:173]: move question to Arrange section & add a Id

**Review by htkisui — COMMENTED** [review]
FIX IT NOW!

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs (hunk 1 above)
Comment (3l5a) [reply QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: Dictateur !!! C'est fait

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs (hunk 2 above)
Comment (3l5a) [reply QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: A vos ordres

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs (hunk 3 above)
Comment (3l5a) [reply QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: Oui chef...........................................

File: QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs (hunk 4 above)
Comment (3l5a) [reply QuikForm.Repository.Contracts/Contracts/IQuestionRepository.cs]: Je crois que j'ai comrpis maontemnatn

File: QuikForm.Repository.Contracts/Exceptions/Questions/QuestionNotFoundException.cs (hunk 5 above)
Comment (3l5a) [reply QuikForm.Repository.Contracts/Exceptions/Questions/QuestionNotFoundException.cs]: 🥇

File: QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs (hunk 7 above)
Comment (3l5a) [reply QuikForm.Repository.Test.Units/Repositories/QuestionRepositoryTests.cs]: i'm the best

**Comment by 3l5a** [comment]
pfff

**Pull request merged**